Lib/configparser.py

```diff
@@ -895,13 +911,19 @@ def write(self, fp, space_around_delimiters=True):
         if self._defaults:
             self._write_section(fp, self.default_section,
                                     self._defaults.items(), d)
+        if UNNAMED_SECTION in self._sections:
+            self._write_section(fp, UNNAMED_SECTION, self._sections[UNNAMED_SECTION].items(), d, unnamed=True)
+
         for section in self._sections:
+            if section is UNNAMED_SECTION:
+                continue
             self._write_section(fp, section,
                                 self._sections[section].items(), d)
 
-    def _write_section(self, fp, section_name, section_items, delimiter):
+    def _write_section(self, fp, section_name, section_items, delimiter, unnamed=False):
         """Write a single section to the specified `fp'."""
-        fp.write("[{}]\n".format(section_name))
+        if not unnamed:
```

Comment:
I think the `unnamed`  parameter is unnecessary and this check can be replaced with `if section_name is not UNNAMED_SECTION`.